### PR TITLE
executor: fix group_concat(a) when a is null

### PR DESCRIPTION
**Pull request by XuHuaiyu**
## What have you changed? (mandatory)
Before this PR, the following will panic:
``` sql
create table t(a int);
insert into t value(null);
select group_concat(a) from t;
```

## What is the type of the changes? (mandatory)

- Bug fix (non-breaking change which fixes an issue)

## How has this PR been tested? (mandatory)
integration test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
no

## Does this PR affect tidb-ansible update? (mandatory)
no
## Does this PR need to be added to the release notes? (mandatory)
no

### CODE REVIEW

File: executor/aggfuncs/func_group_concat.go [hunk 1]
```diff
@@ -103,8 +103,11 @@ func (e *groupConcat) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup [
 			p.buffer.WriteString(e.sep)
 		}
 	}
-	p.buffer.Truncate(p.buffer.Len() - len(e.sep))
-	return e.truncatePartialResultIfNeed(sctx, p.buffer)
+	if p.buffer != nil {
```
Comment (shenli) [inline executor/aggfuncs/func_group_concat.go]: When will the p.buffer be nil?

Comment (XuHuaiyu) [reply executor/aggfuncs/func_group_concat.go]: As the test case shows, if all the value of the arg is null, 
p.buffer will be nil.

File: executor/aggfuncs/func_group_concat.go [hunk 2]
```diff
@@ -103,8 +103,11 @@ func (e *groupConcat) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup [
 			p.buffer.WriteString(e.sep)
 		}
 	}
-	p.buffer.Truncate(p.buffer.Len() - len(e.sep))
```
Comment (lysu) [inline executor/aggfuncs/func_group_concat.go]: `group_concat distinct` doesn't panic but got the wrong result.

```
mysql> select group_concat(distinct a) from t;
+--------------------------+
| group_concat(distinct a) |
+--------------------------+
|                          |
+--------------------------+
1 row in set (0.00 sec)
```
in mysql return null

Comment (zhexuany) [reply executor/aggfuncs/func_group_concat.go]: I can reproduce this via queries provided in this PR. Maybe you need double-check?

**Comment by XuHuaiyu** [comment]
PTAL @zhexuany @zz-jason

**Pull request merged**